File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/VB/Customer/CustomerWorkflow.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules
Imports System.Xml

Namespace Microsoft.WorkflowServices.Samples
	Public NotInheritable Partial Class CustomerWorkflow
		Inherits SequentialWorkflowActivity
		Public Sub New()
			InitializeComponent()
		End Sub

		Public order As New PurchaseOrder()
		Public supplierAck As String = Nothing
        Public contextToSend As IDictionary(Of String, String) = Nothing

		Private Sub PrepareOrder(ByVal sender As Object, ByVal e As SendActivityEventArgs)
			Me.order.Amount = 1000
			Me.order.OrderId = 1234
			Me.contextToSend = Me.ReceiveOrderDetails.Context
		End Sub

		Public orderDetails As New PurchaseOrder()
		Public shippingQuote As New ShippingQuote()
		Public customerAck As String = Nothing

		Private Sub ReviewOrder(ByVal sender As Object, ByVal e As EventArgs)
			Me.customerAck = "Order Details Received"
			Console.ForegroundColor = ConsoleColor.Green
			Console.WriteLine("Order Accepted.")
            Console.WriteLine("Cost: $" + Me.shippingQuote.ShippingCost.ToString())
            Console.WriteLine("ShipDate: " + Me.shippingQuote.EstimatedShippingDate.ToString())
			Console.ResetColor()
		End Sub

		Private Sub DisplayStatus(ByVal sender As Object, ByVal e As EventArgs)
			Console.WriteLine(Me.supplierAck)
		End Sub
	End Class

End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/CS/Customer/CustomerWorkflow.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections.Generic;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Xml;

namespace Microsoft.WorkflowServices.Samples
{
	public sealed partial class CustomerWorkflow : SequentialWorkflowActivity
	{
        public CustomerWorkflow()
		{
			InitializeComponent();
		}

        public PurchaseOrder order = new PurchaseOrder();
        public string supplierAck = default(string);
        public IDictionary<string, string> contextToSend = default(IDictionary<string, string>);

        private void PrepareOrder(object sender, SendActivityEventArgs e)
        {
            this.order.Amount = 1000;
            this.order.OrderId = 1234;
            this.contextToSend = this.ReceiveOrderDetails.Context;
        }

        public PurchaseOrder orderDetails = new PurchaseOrder();
        public ShippingQuote shippingQuote = new ShippingQuote();
        public string customerAck = default(string);

        private void ReviewOrder(object sender, EventArgs e)
        {
            this.customerAck = "Order Details Received";
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("Order Accepted.");
            Console.WriteLine("Cost: $" + this.shippingQuote.ShippingCost);
            Console.WriteLine("ShipDate: " + this.shippingQuote.EstimatedShippingDate);
            Console.ResetColor();
        }

        private void DisplayStatus(object sender, EventArgs e)
        {
            Console.WriteLine(this.supplierAck);
        }
	}

}
